backend/fms_core/models/sample_lineage.py

```diff
@@ -31,16 +31,17 @@ def add_error(field: str, error: str):
                 parent_derived = self.parent.derived_samples.get()
             except MultipleObjectsReturned:
                 add_error("derived_sample", "Extraction child and/or parent has more than one derived sample.")
-                
-            if not child_derived.sample_kind.is_extracted:
-                add_error("sample_kind", "Extracted sample need to be a type of Nucleic Acid.")
-            if parent_derived.sample_kind.is_extracted:
-                add_error("extracted_from",
-                          "Extraction process cannot be run on samples of extracted kinds like DNA and RNA.")
-            if not child_derived.tissue_source:
-                add_error("tissue_source", "Extracted sample need to have a tissue source.")
-            elif child_derived.tissue_source != parent_derived.sample_kind:
-                add_error("tissue_source", "Extracted sample tissue_source must match parent sample_kind.")
+
+            if not errors:
+                if not child_derived.sample_kind.is_extracted:
+                    add_error("sample_kind", "Extracted sample need to be a type of Nucleic Acid.")
+                if parent_derived.sample_kind.is_extracted:
+                    add_error("extracted_from",
+                              "Extraction process cannot be run on samples of extracted kinds like DNA and RNA.")
+                if not child_derived.tissue_source:
+                    add_error("tissue_source", "Extracted sample need to have a tissue source.")
+                elif child_derived.tissue_source != parent_derived.sample_kind:
+                    add_error("tissue_source", "Extracted sample tissue_source must match parent sample_kind.")
         elif protocol_name == "Transfer":
```

Comment:
We would need to add validation here for "Sample Pooling" too. We should check that parent.derived_sample are a subset of the child.derived_sample. Also "Experiment Run" should share the current validation with "Transfer".